tests/unit/Restoration.spec.ts

```diff
@@ -0,0 +1,373 @@
+        addressCity: 'Victoria',
+        addressCountry: 'CA',
+        addressRegion: 'BC',
+        deliveryInstructions: null,
+        id: 66666,
+        postalCode: 'V6V 6V6',
+        streetAddress: '666 Main Street',
+        streetAddressAdditional: null
+      },
+      officer: {},
+      role: 'applicant'
+    }
+  ]
+}
+const stateFiling = {
+  header: {
+    name: 'restoration',
+    status: 'COMPLETED'
+  },
+  business: {
+    identifier: 'BC1234567',
+    legalName: '1234567 B.C. LTD.',
+    legalType: 'BEN'
+  },
+  restoration: {
+    expiry: '2023-01-31',
+    type: 'limitedRestoration'
+  }
+}
+const filingFees = {
+  filingFees: 20.0
+}
+
+// init session storage variables
+sessionStorage.setItem('PAY_API_URL', 'https://pay.api.url/')
+sessionStorage.setItem('KEYCLOAK_TOKEN', 'keycloak-token') // anything non-falsy
+
+describe('Restoration component - edit page', () => {
+  const { assign } = window.location
+  let wrapper: any
+
+  beforeAll(async () => {
+    // init store
+    store.state.stateModel.summaryMode = false
+    store.state.stateModel.validationFlags.appValidate = false
+    store.state.stateModel.tombstone.businessId = 'BC1234567' // normally set in App.vue
+    store.state.stateModel.tombstone.keycloakRoles = ['staff'] // normally set in App.vue
+
+    // mock the window.location.assign function
+    delete window.location
+    window.location = { assign: jest.fn() } as any
+
+    // mock services function
+    jest.spyOn((LegalServices as any), 'fetchFilingById').mockImplementation(
+      () => Promise.resolve(filing))
+    jest.spyOn((LegalServices as any), 'fetchBusinessInfo').mockImplementation(
+      () => Promise.resolve(businessInfo))
+    jest.spyOn((AuthServices as any), 'fetchAuthInfo').mockImplementation(
+      () => Promise.resolve(authInfo))
+    jest.spyOn((LegalServices as any), 'fetchAddresses').mockImplementation(
+      () => Promise.resolve(addresses))
+    jest.spyOn((LegalServices as any), 'fetchNameTranslations').mockImplementation(
+      () => Promise.resolve(nameTranslations))
+    jest.spyOn((LegalServices as any), 'fetchDirectors').mockImplementation(
+      () => Promise.resolve(directors))
+    jest.spyOn((LegalServices as any), 'fetchFiling').mockImplementation(
+      () => Promise.resolve(stateFiling))
+    jest.spyOn((PayServices as any), 'fetchFilingFees').mockImplementation(
+      () => Promise.resolve(filingFees))
+
+    // create a Local Vue and install router on it
+    const localVue = createLocalVue()
+    localVue.use(VueRouter)
+    const router = mockRouter.mock()
+    await router.push({ name: 'restoration', query: { 'restoration-id': '1234' } })
+    wrapper = mount(Restoration, { localVue, store, router, vuetify })
+
+    // enable filing and wait for all queries to complete
+    await wrapper.setProps({ appReady: true })
+    await flushPromises()
+  })
+
+  afterAll(() => {
+    window.location.assign = assign
+    jest.restoreAllMocks()
```

Comment:
I would have like to combine the 3 "describe" sections in this file but I was unable to correctly restore/clear/reset the mocks in a reasonable amount of time. So there's some code duplication in here, oh well.